src/playground/sample_data_tooling/sample_data_ingestor/sample_data_ingestor.py

```diff
@@ -0,0 +1,336 @@
+        - current_date: The date at which entries are generated
+        - max_bulk_size: The max amount in bytes of a bulk call
+
+    Raises:
+        - TypeError: client should be an OpenSearch Python client object
+        - TypeError: data_template should be a filename string, a JSON string, or dict
+        - TypeError: index_name should be a string
+        - TypeError: file_provided should be a boolean flag
+        - TypeError: mapping should be a boolean flag, not a data template
+        - ValueError: number should be a positive integer
+        - ValueError: chunk should be a positive integer
+        - TypeError: timestamp should be the field name that contains the timestamp, not the actual unix timestamp
+        - ValueError: minutes should be a positive integer
+        - ValueError: current_date should be a datetime object
+        - ValueError: max_bulk_size should be a positive integer
+        - TypeError: anomaly_detection_trend should be a list of config dicts
+    """
+    if client and (not isinstance(client, OpenSearch)):
+        raise TypeError("client should be an OpenSearch Python client object")
+    if data_template and type(data_template) is not str and type(data_template) is not dict:
+        raise TypeError("data_template should be a filename string, a JSON string, or dict")
+    if index_name and type(index_name) is not str:
+        raise TypeError("index_name should be a string")
+    if file_provided and type(file_provided) is not bool:
+        raise TypeError("file_provided should be a boolean flag")
+    if mapping and type(mapping) is not bool:
+        raise TypeError("mapping should be a boolean flag, not a data template")
+    if number and type(number) is not int or (type(number) is int and number < 0):
+        raise ValueError("number should be a positive integer")
+    if chunk and type(chunk) is not int or (type(chunk) is int and chunk < 0):
+        raise ValueError("chunk should be a positive integer")
+    if timestamp and type(timestamp) is not str:
+        raise TypeError("timestamp should be the field name that contains the timestamp, not the actual unix timestamp")
+    if minutes and type(minutes) is not int or (type(minutes) is int and minutes < 0):
+        raise ValueError("minutes should be a positive integer")
+    if current_date and type(current_date) is not datetime:
+        raise ValueError("current_date should be a datetime object")
+    if max_bulk_size and type(max_bulk_size) is not int or (type(max_bulk_size) is int and max_bulk_size < 0):
+        raise ValueError("max_bulk_size should be a positive integer")
+    if anomaly_detection_trend and type(anomaly_detection_trend) is not list:
+        raise TypeError("anomaly_detection_trend should be a list of config dicts")
+
+
+def build_request_body(index_name, 
+    file_provided, 
+    timestamp, 
```

Comment:
same as above, please remove all the unnecessary whitespace.

Reply:
Done, all trailing whitespace removed